File: 4.Sorting/2.BubbleSort/solution.js

function bubbleSort(arr){
    let isSwapped = false;
    for(let i = 0; i < arr.length; i++){
        for(let j = 0; j < arr.length-i-1; j++){
            if(arr[j] > arr[j+1]){
                let temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
                isSwapped = true;
            }
        }
    }
    if(isSwapped == true) return;
}

//const arr = [4,1,3,2,9];
const arr = [1,2,3,4,9];
bubbleSort(arr);
console.log(arr);

File: 4.Sorting/2.BubbleSort/solution.java
public class solution {
 
    
    public static void bubbleSort(int nums[]){
        
        for(int i = 0; i < nums.length; i++){
            for(int j = 0; j < nums.length-i-1; j++){
                if(nums[j] > nums[j+1]){
                    int temp = nums[j];
                    nums[j] = nums[j+1];
                    nums[j+1] = temp;
                }
            }
        }
    }

    public static void main(String args[]){
        int arr[] = {5,7,1,3,0};
        bubbleSort(arr);
        for(int i : arr){
            System.out.print(i+" ");
        }
    }
}
